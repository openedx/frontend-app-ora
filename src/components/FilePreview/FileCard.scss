@import "@edx/paragon/scss/core/core";

.file-card {
  margin: map-get($spacers, 1) 0;

  .file-card-title {
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
  }
}

.image-renderer {
  width: 100%;
  height: auto;
}

// .pdf-renderer {
//   .react-pdf__Page__canvas {
//     width: 100% !important;
//     height: auto !important;
//   }
// }

.txt-renderer {
  white-space: pre-wrap;
}

@include media-breakpoint-down(sm) {
  .file-card-title {
    width: calc(map-get($container-max-widths, "sm")/2);
  }
}