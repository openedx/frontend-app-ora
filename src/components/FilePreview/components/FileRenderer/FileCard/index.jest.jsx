import { shallow } from '@edx/react-unit-test-utils';

import FileCard from './index';

describe('<FileCard />', () => {
  const props = {
    file: {
      fileName: 'fileName',
    },
    children: <div>children</div>,
    defaultOpen: true,
  };

  it('renders default', () => {
    const wrapper = shallow(<FileCard {...props} />);
    expect(wrapper.snapshot).toMatchSnapshot();

    expect(wrapper.instance.findByTestId('input').length).toBe(1);
    expect(wrapper.instance.findByTestId('feedback').length).toBe(1);
  });
});