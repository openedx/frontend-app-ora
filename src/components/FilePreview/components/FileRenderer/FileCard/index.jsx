import React from 'react';
import PropTypes from 'prop-types';

import { Card, Collapsible } from '@edx/paragon';

import './FileCard.scss';

/**
 * <FileCard />
 */
const FileCard = ({ file, children, defaultOpen }) => (
  <Card className="file-card" key={file.fileName}>
    <Collapsible
      className="file-collapsible"
      defaultOpen={defaultOpen}
      title={<h3 className="file-card-title">{file.fileName}</h3>}
    >
      <div className="preview-panel">
        {children}
      </div>
    </Collapsible>
  </Card>
);
FileCard.propTypes = {
  file: PropTypes.shape({ fileName: PropTypes.string.isRequired }).isRequired,
  children: PropTypes.node.isRequired,
  defaultOpen: PropTypes.bool.isRequired,
};

export default FileCard;
