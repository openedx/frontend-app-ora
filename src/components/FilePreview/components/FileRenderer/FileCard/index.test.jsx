import { shallow } from '@edx/react-unit-test-utils';

import FileCard from './index';

describe('<FileCard />', () => {
  const props = {
    file: {
      fileName: 'fileName',
    },
    children: <div>children</div>,
    defaultOpen: true,
  };

  it('renders default', () => {
    const wrapper = shallow(<FileCard {...props} />);
    expect(wrapper.snapshot).toMatchSnapshot();

    expect(wrapper.instance.findByType('Collapsible')).toHaveLength(1);
  });
});
