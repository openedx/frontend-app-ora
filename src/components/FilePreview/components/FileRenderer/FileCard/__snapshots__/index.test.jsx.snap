// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileCard /> renders default 1`] = `
<Card
  className="file-card"
  key="fileName"
>
  <Collapsible
    className="file-collapsible"
    defaultOpen={true}
    title={
      <h3
        className="file-card-title"
      >
        fileName
      </h3>
    }
  >
    <div
      className="preview-panel"
    >
      <div>
        children
      </div>
    </div>
  </Collapsible>
</Card>
`;
